src/base/QXmppMessage.cpp

```diff
@@ -1442,7 +1474,22 @@ bool QXmppMessage::parseExtension(const QDomElement &element, QXmpp::SceMode sce
 ///
 void QXmppMessage::serializeExtensions(QXmlStreamWriter *writer, QXmpp::SceMode sceMode, const QString &baseNamespace) const
 {
+    const auto writeTextElement = [writer, &baseNamespace](const QString &tagName, const QString &text) {
+        if (!text.isEmpty()) {
+            writer->writeStartElement(tagName);
+            if (!baseNamespace.isNull()) {
+                writer->writeDefaultNamespace(baseNamespace);
+            }
+            writer->writeCharacters(text);
+            writer->writeEndElement();
+        }
+    };
+
     if (sceMode & QXmpp::ScePublic) {
+        if (sceMode == QXmpp::ScePublic) {
+            writeTextElement(QStringLiteral("body"), d->e2eeFallbackBody);
```

Comment:
writeTextElement() is only for private elements, use helperToXmlAddTextElement()

Reply:
I used `writeTextElement()` instead because it should not be an empty element if the fallback body is empty.